java/impl/aws/src/main/java/co/worklytics/psoxy/Handler.java

```diff
@@ -1,33 +1,29 @@
 package co.worklytics.psoxy;
 
 import co.worklytics.psoxy.aws.DaggerAwsContainer;
-import co.worklytics.psoxy.aws.LambdaRequest;
+import co.worklytics.psoxy.aws.request.APIGatewayV2HTTPEventRequestAdapter;
 import co.worklytics.psoxy.gateway.HttpEventResponse;
 import co.worklytics.psoxy.gateway.impl.CommonRequestHandler;
 import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayV2HTTPEvent;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayV2HTTPResponse;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import lombok.*;
+import lombok.SneakyThrows;
 import lombok.extern.java.Log;
+import org.apache.commons.lang3.exception.ExceptionUtils;
 
 import javax.inject.Inject;
-import java.util.Map;
 
 @Log
-public class Handler implements com.amazonaws.services.lambda.runtime.RequestHandler<Map<String, Object>, String> {
+public class Handler implements com.amazonaws.services.lambda.runtime.RequestHandler<APIGatewayV2HTTPEvent, APIGatewayV2HTTPResponse> {
 
 
     @Inject
     CommonRequestHandler requestHandler;
 
-    @Inject
-    ObjectMapper objectMapper;
-
-    //TODO: improve this
-    //   - change this to directly take LambdaRequest as it's parameter type
-    //   - or to take InputStream, and then parse that with jackson (AWS lambda docs implies this is supported, but doesn't work)
     @SneakyThrows
     @Override
-    public String handleRequest(Map<String, Object> request, Context context) {
+    public APIGatewayV2HTTPResponse handleRequest(APIGatewayV2HTTPEvent httpEvent, Context context) {
```

Comment:
Directly use their http events, just created `APIGatewayV2HTTPEventRequestAdapter` to the interface the `CommonHandler` understands and back.